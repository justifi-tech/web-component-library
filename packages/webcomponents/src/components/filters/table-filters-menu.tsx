import { Component, Listen, Prop, State, Watch, h } from '@stencil/core';
import { Button } from '../../ui-components/button';
import { hasFilters } from './utils';
import { dropdownMenu } from '../../styles/parts';

@Component({
  tag: 'table-filters-menu'
})
export class TableFiltersMenu {
  @State() anchorButton: HTMLElement;
  @State() menu: HTMLElement;
  @State() showClearButton: boolean = false;

  @Prop() params: any;
  @Prop() clearParams: () => void;

  @Watch('params')
  watchParamsHandler(newValue: any) {
    this.showClearButton = hasFilters(newValue);
  }

  @Listen('showEvent')
  show() {
    this.menu.classList.add('show');
  }

  @Listen('hideEvent')
  hide() {
    this.menu.classList.remove('show');
  }

  componentWillLoad() {
    this.showClearButton = hasFilters(this.params);
  }

  render() {
    return (
      <div class="d-flex gap-2">
        <div class="dropdown pb-3">
          <Button
            variant="primary"
            type="button dropdown-toggle"
            ref={(el) => (this.anchorButton = el)}
            class="btn btn-primary dropdown-toggle"
            data-test-id="open-filters-button"
            data-bs-toggle="dropdown"
          >
            {"Filters"}
          </Button>
          <custom-popper
            offset={[20, 10]}
            strategy="fixed"
            anchorRef={this.anchorButton}
            trigger="click"
          >
            <form
              class="dropdown-menu"
              part={dropdownMenu}
              data-test-id="filter-menu"
              ref={(el) => (this.menu = el)}
              onSubmit={(e) => e.preventDefault()}
            >
              {/* Render filter inputs here */}
              <slot />
            </form>
          </custom-popper>
        </div>
        <div>
          <Button
            variant="secondary"
            onClick={() => this.clearParams()}
            hidden={!this.showClearButton}
            data-test-id="clear-filters-button"
          >
            {"Clear Filters"}
          </Button>
        </div>
      </div>
    );
  }
}
