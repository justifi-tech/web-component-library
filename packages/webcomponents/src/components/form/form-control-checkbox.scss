@import 'reboot';
@import 'forms';

:host {
  @include host-base-styles;
  display: block;
}

// input[type='checkbox'][disabled] {
//   color: var(--jfi-form-control-disabled-color);
//   opacity: 0.5;
//   cursor: not-allowed;
// }

*::part(input[type='checkbox']) {
  margin: var(--jfi-form-control-margin);
  padding: var(--jfi-form-control-padding);
}