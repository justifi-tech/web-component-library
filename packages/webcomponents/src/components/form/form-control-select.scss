@import 'reboot';
@import 'forms';

:host {
  @include host-base-styles;
  display: block;
}

select[disabled] {
  background-color: #e9ecef;
  opacity: 0.5;
  cursor: not-allowed;
}
