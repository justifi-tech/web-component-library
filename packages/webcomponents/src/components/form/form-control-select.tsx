import {
  Component,
  Host,
  h,
  Prop,
  Event,
  EventEmitter,
  Watch,
} from '@stencil/core';

@Component({
  tag: 'form-control-select'
})
export class SelectInput {
  selectElement!: HTMLSelectElement;

  @Prop() label: string;
  @Prop() name: any;
  @Prop() helpText?: string;
  @Prop() errorText?: string;
  @Prop() defaultValue: string;
  @Prop() inputHandler: (name: string, value: string) => void;
  @Prop() options: { label: string; value: string }[];
  @Prop() disabled: boolean;

  @Event() formControlInput: EventEmitter<any>;
  @Event() formControlBlur: EventEmitter<any>;

  @Watch('defaultValue')
  handleDefaultValueChange(newValue: string) {
    this.updateInput(newValue);
  }

  updateInput(newValue: any) {
    this.selectElement.value = newValue;
  }

  handleFormControlInput(event: any) {
    const target = event.target;
    const name = target.getAttribute('name');
    this.inputHandler(name, target.value);
    this.formControlInput.emit(target.value);
  }

  componentDidLoad() {
    this.updateInput(this.defaultValue);
  }

  render() {
    return (
      <Host exportparts="label,input,input-invalid">
        <div class="form-group d-flex flex-column">
          <label part="label" class="form-label" htmlFor={this.name}>
            {this.label}
          </label>
          <select
            ref={el => (this.selectElement = el as HTMLSelectElement)}
            id={this.name}
            name={this.name}
            onInput={(event: any) => this.handleFormControlInput(event)}
            onBlur={() => this.formControlBlur.emit()}
            part={`input ${this.errorText ? 'input-invalid' : ''}`}
            class={this.errorText ? 'form-select is-invalid' : 'form-select'}
            disabled={this.disabled}
          >
            {this.options?.map(option => (
              <option value={option.value}>{option.label}</option>
            ))}
          </select>
          <form-control-help-text helpText={this.helpText} />
          <form-control-error-text errorText={this.errorText} />
        </div>
      </Host>
    );
  }
}
