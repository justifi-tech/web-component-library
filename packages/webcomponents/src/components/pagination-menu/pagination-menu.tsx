import { Component, h, Prop } from '@stencil/core';
import { ExtendedPagingDefaults, ExtendedPagingInfo } from '../../api/Pagination';
import { StyledHost } from '../../ui-components';

export interface PaginationProps {
  paging: ExtendedPagingInfo;
  params?: {};
}

@Component({
  tag: 'pagination-menu'
})

export class PaginationMenu {
  @Prop() paging: PaginationProps['paging'] = ExtendedPagingDefaults;
  @Prop() params: PaginationProps['params'] = {};

  render() {
    return (
      <StyledHost exportparts="pagination-buttons-container">
        <nav
          aria-label="Table pagination"
          class="d-flex justify-content-end gap-3"
          part="pagination-bar"
        >
          <button
            class="page-link"
            onClick={(e) => {
              e.preventDefault();
              this.paging.handleClickPrevious(this.paging.start_cursor);
            }}
            part={`button-link ${this.paging.has_next ? '' : ' button-disabled'}`}
          >
            <span class="me-1" part="page-arrow">&laquo;</span>
            <span part="page-button-text">Previous</span>
          </button>
          <button
            class="page-link"
            onClick={(e) => {
              e.preventDefault();
              this.paging.handleClickNext(this.paging.end_cursor);
            }}
            part={`button-link ${this.paging.has_next ? '' : ' button-disabled'}`}
          >
            <span part="page-button-text">Next</span>
            <span class="ms-1" part="page-arrow">&raquo;</span>
          </button>
        </nav>
      </StyledHost >
    );
  }
}
