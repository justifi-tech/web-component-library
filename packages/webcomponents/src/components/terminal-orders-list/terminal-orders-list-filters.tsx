import { Component, h, Prop } from '@stencil/core';
import { filterParams, propsParams, clearParams } from './terminal-orders-list-params-state';
import { StyledHost } from '../../ui-components';
import { TerminalOrderStatus, TerminalOrderType } from '../../api';
import { convertToLocal, convertToUTC } from '../../utils/utils';
import {
  terminalOrdersListFilterMenu,
  orderStatusTerminalOrdersListFilterParam,
  orderTypeTerminalOrdersListFilterParam,
  createdAfterTerminalOrdersListFilterParam,
  createdBeforeTerminalOrdersListFilterParam
} from '../../styles/parts';

@Component({
  tag: 'justifi-terminal-orders-list-filters',
  shadow: true
})
export class TerminalOrdersListFilters {
  @Prop() orderStatus?: TerminalOrderStatus;
  @Prop() orderType?: TerminalOrderType;
  @Prop() createdAfter?: string;
  @Prop() createdBefore?: string;

  componentWillLoad() {
    const propsToSet = {
      order_status: this.orderStatus,
      order_type: this.orderType,
      created_after: this.createdAfter,
      created_before: this.createdBefore
    };

    Object.entries(propsToSet).forEach(([key, value]) => {
      if (value) {
        propsParams[key] = value;
      }
    });
  }

  setParamsOnChange = (name: string, value: string) => {
    filterParams[name] = value;
  };

  handleDateInput = (name: string, value: string) => {
    const utcDate = convertToUTC(value, { setExactTime: true });
    this.setParamsOnChange(name, utcDate);
  }

  get terminalOrderStatusOptions(): { label: string, value: any | '' }[] {
    return [
      { label: 'All', value: '' },
      { label: 'Created', value: TerminalOrderStatus.created },
      { label: 'Completed', value: TerminalOrderStatus.completed },
      { label: 'Submitted', value: TerminalOrderStatus.submitted },
      { label: 'In Progress', value: TerminalOrderStatus.in_progress },
      { label: 'On Hold', value: TerminalOrderStatus.on_hold },
      { label: 'Canceled', value: TerminalOrderStatus.canceled }
    ]
  }

  get terminalOrderTypeOptions(): { label: string, value: TerminalOrderType | '' }[] {
    return [
      { label: 'All', value: '' },
      { label: 'Boarding Only', value: TerminalOrderType.boardingOnly },
      { label: 'Boarding Shipping', value: TerminalOrderType.boardingShipping },
    ]
  }

  render() {
    const filterMenuParams = { ...filterParams }

    return (
      <StyledHost>
        <table-filters-menu params={filterMenuParams} clearParams={clearParams} part={terminalOrdersListFilterMenu}>
          <div class="grid-cols-2 gap-3 p-1">
            <div class="p-2">
              <form-control-select
                name="order_status"
                label="Order Status"
                options={this.terminalOrderStatusOptions}
                inputHandler={this.setParamsOnChange}
                defaultValue={this.orderStatus || filterParams.order_status || ''}
                disabled={!!this.orderStatus}
                part={orderStatusTerminalOrdersListFilterParam}
              />
            </div>
            <div class="p-2">
              <form-control-select
                name="order_type"
                label="Order Type"
                options={this.terminalOrderTypeOptions}
                inputHandler={this.setParamsOnChange}
                defaultValue={this.orderType || filterParams.order_type || ''}
                disabled={!!this.orderType}
                part={orderTypeTerminalOrdersListFilterParam}
              />
            </div>
            <div class="p-2">
              <form-control-date
                name="created_after"
                label="Created After"
                inputHandler={this.handleDateInput}
                defaultValue={
                  convertToLocal(this.createdAfter, { showInputDateTime: true }) ||
                  convertToLocal(filterParams.created_after, { showInputDateTime: true })
                }
                showTime
                disabled={!!this.createdAfter}
                part={createdAfterTerminalOrdersListFilterParam}
              />
            </div>
            <div class="p-2">
              <form-control-date
                name="created_before"
                label="Created Before"
                inputHandler={this.handleDateInput}
                defaultValue={
                  convertToLocal(this.createdBefore, { showInputDateTime: true }) ||
                  convertToLocal(filterParams.created_before, { showInputDateTime: true })
                }
                showTime
                disabled={!!this.createdBefore}
                part={createdBeforeTerminalOrdersListFilterParam}
              />
            </div>
          </div>
        </table-filters-menu>
      </StyledHost>
    );
  }
}
