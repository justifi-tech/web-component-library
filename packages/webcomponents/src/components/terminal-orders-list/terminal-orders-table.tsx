import { h } from '@stencil/core';
import { getAlternateTableCellPart, tableHeadCell } from '../../styles/parts';

export const defaultColumnsKeys = 'id,status';

export const terminalOrdersTableColumns = {
  id: () => (
    <th part={tableHeadCell} scope="col" title="The ID of the terminal order">
      ID
    </th>
  ),
  status: () => (
    <th part={tableHeadCell} scope="col" title="The status of the terminal order">
      Status
    </th>
  ),
}

export const terminalOrdersTableCells = {
  id: (terminalOrder, index) => (
    <td part={getAlternateTableCellPart(index)}>{terminalOrder.id}</td>
  ),
  status: (terminalOrder, index) => (
    <td part={getAlternateTableCellPart(index)}>{terminalOrder.status}</td>
  ),
};
