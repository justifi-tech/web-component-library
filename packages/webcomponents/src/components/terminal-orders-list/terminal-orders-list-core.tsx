import { Component, h, Prop, State, Watch, Event, EventEmitter } from '@stencil/core';
import { Table } from '../../utils/table';
import { table, tableCell } from '../../styles/parts';
import { TableEmptyState, TableErrorState, TableLoadingState } from '../../ui-components';
import { pagingDefaults, PagingInfo } from '../../api';
import { terminalOrdersTableCells, terminalOrdersTableColumns } from './terminal-orders-table';
import { TerminalOrder } from '../../api';
import mockTerminalOrders from '../../../../../mockData/mockTerminalOrdersListSuccess.json';

@Component({
  tag: 'terminal-orders-list-core',
})

export class TerminalOrdersListCore {
  @Prop() getTerminalOrders: Function;
  @Prop() columns: string;

  @State() terminalOrders: TerminalOrder[] = mockTerminalOrders.data as TerminalOrder[];
  @State() terminalOrdersTable: Table<TerminalOrder>;
  @State() loading: boolean = true;
  @State() errorMessage: string;
  @State() paging: PagingInfo = pagingDefaults;
  @State() pagingParams: any = {};

  @Watch('pagingParams')
  @Watch('getTerminalOrders')
  @Watch('columns')
  updateOnPropChange() {
    this.fetchData();
  }

  @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<any>;

  componentWillLoad() {
    this.terminalOrdersTable = new Table<TerminalOrder>(this.terminalOrders, this.columns, terminalOrdersTableColumns, terminalOrdersTableCells);
    if (this.getTerminalOrders) {
      this.fetchData();
    }
  }

  fetchData(): void {
    this.loading = true;
    
    this.getTerminalOrders({
      params: {},
      onSuccess: async ({ terminalOrders, pagingInfo }) => {
        this.terminalOrders = terminalOrders;
        this.paging = pagingInfo;
        this.terminalOrdersTable.collectionData = this.terminalOrders;
      },
      onError: ({ error, code, severity }) => {
        this.errorMessage = error;
        this.errorEvent.emit({ error, code, severity });
      },
      final: () => { this.loading = false },
    });
  }

  handleClickPrevious = (beforeCursor: string) => {
    this.pagingParams = { before_cursor: beforeCursor };
  };

  handleClickNext = (afterCursor: string) => {
    this.pagingParams = { after_cursor: afterCursor };
  };

  rowClickHandler = (e) => {
    const clickedOrderId = e.target.closest('tr').dataset.rowEntityId;
    if (!clickedOrderId) return;
    console.log('clickedOrderId', clickedOrderId);
  };

  get entityId() {
    return this.terminalOrders.map((terminalOrder) => terminalOrder.id);
  }

  get showEmptyState() {
    return !this.loading && !this.errorMessage && this.terminalOrdersTable.rowData.length < 1;
  }

  get showErrorState() {
    return !this.loading && !!this.errorMessage;
  }

  get showRowData() {
    return !this.showEmptyState && !this.showErrorState && !this.loading;
  }

  render() {
    return (
      <div>
        <div class="table-wrapper">
          <table class="table table-hover" part={table}>
            <thead class="table-head sticky-top">
              <tr class="table-light text-nowrap">
                {this.terminalOrdersTable.columnData.map((column) => column)}
              </tr>
            </thead>
            <tbody class="table-body">
              <TableLoadingState
                columnSpan={this.terminalOrdersTable.columnKeys.length}
                isLoading={this.loading}
              />
              <TableEmptyState
                isEmpty={this.showEmptyState}
                columnSpan={this.terminalOrdersTable.columnKeys.length}
              />
              <TableErrorState
                columnSpan={this.terminalOrdersTable.columnKeys.length}
                errorMessage={this.errorMessage}
              />
              {this.showRowData && this.terminalOrdersTable.rowData.map((data, index) => (
                <tr
                  data-test-id="table-row"
                  data-row-entity-id={this.entityId[index]}
                  onClick={this.rowClickHandler}
                >
                  {data}
                </tr>
              ))}
            </tbody>
            {this.paging && (
              <tfoot class="sticky-bottom">
                <tr class="table-light align-middle">
                  <td part={tableCell} colSpan={this.terminalOrdersTable.columnData.length}>
                    <pagination-menu
                      paging={{
                        ...this.paging,
                        handleClickPrevious: this.handleClickPrevious,
                        handleClickNext: this.handleClickNext,
                      }}
                    />
                  </td>
                </tr>
              </tfoot>
            )}
          </table>
        </div>
      </div>
    );
  }
}
