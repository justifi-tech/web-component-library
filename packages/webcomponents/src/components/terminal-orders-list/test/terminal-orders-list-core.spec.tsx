jest.mock('../../../ui-components/styled-host/styled-host.css', () => '');

import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { TerminalOrdersListCore } from '../terminal-orders-list-core';
import { PaginationMenu } from '../../pagination-menu/pagination-menu';
import mockSuccessResponse from '../../../../../../mockData/mockTerminalOrdersListSuccess.json';
import { IApiResponseCollection, ITerminalOrder } from '../../../api';
import { makeGetTerminalOrders } from '../../../actions/terminal/get-terminal-orders';
import { TerminalOrdersListFilters } from '../terminal-orders-list-filters';
import { TableFiltersMenu } from '../../../ui-components/filters/table-filters-menu';
import { SelectInput } from '../../../ui-components/form/form-control-select';
import { defaultColumnsKeys } from '../terminal-orders-table';

const mockOrdersResponse = mockSuccessResponse as IApiResponseCollection<ITerminalOrder[]>;
const components = [TerminalOrdersListCore, PaginationMenu, TableFiltersMenu, TerminalOrdersListFilters, SelectInput];

describe('terminal-orders-list-core', () => {

  it('renders properly with fetched data', async () => {
    const mockTerminalsService = {
      fetchTerminalOrders: jest.fn().mockResolvedValue(mockOrdersResponse),
    };

    const getTerminalOrders = makeGetTerminalOrders({
      id: '123',
      authToken: '123',
      service: mockTerminalsService,
      apiOrigin: 'http://localhost:3000'
    });

    const page = await newSpecPage({
      components: components,
      template: () => <terminal-orders-list-core getTerminalOrders={getTerminalOrders} columns={defaultColumnsKeys} />,
    });

    await page.waitForChanges();

    expect(page.rootInstance.terminalOrders[0]).toEqual(expect.objectContaining({ sub_account_id: mockOrdersResponse.data[0].sub_account_id }));
    const rows = page.root.querySelectorAll('[data-test-id="table-row"]');
    expect(rows.length).toBe(mockOrdersResponse.data.length);
    expect(mockTerminalsService.fetchTerminalOrders).toHaveBeenCalled();
    expect(page.root).toMatchSnapshot();
  });

  it('displays an error state on failed data fetch', async () => {
    const mockService = {
      fetchTerminalOrders: jest.fn().mockRejectedValue(new Error('Fetch error'))
    };

    const getTerminalOrders = makeGetTerminalOrders({
      id: 'some-id',
      authToken: 'some-auth-token',
      service: mockService,
      apiOrigin: 'http://localhost:3000'
    });

    const page = await newSpecPage({
      components: components,
      template: () => <terminal-orders-list-core getTerminalOrders={getTerminalOrders} columns={defaultColumnsKeys} />,
    });

    await page.waitForChanges();

    expect(page.rootInstance.errorMessage).toBe('Fetch error');
    expect(page.root).toMatchSnapshot();
  });

  it('emits click-event event on row click', async () => {
    const mockTerminalsService = {
      fetchTerminalOrders: jest.fn().mockResolvedValue(mockOrdersResponse),
    };

    const getTerminalOrders = makeGetTerminalOrders({
      id: '123',
      authToken: '123',
      service: mockTerminalsService,
      apiOrigin: 'http://localhost:3000'
    });

    const page = await newSpecPage({
      components: components,
      template: () => <terminal-orders-list-core getTerminalOrders={getTerminalOrders} columns={defaultColumnsKeys} />,
    });

    await page.waitForChanges();

    const firstRow = page.root.querySelector('[data-test-id="table-row"]') as HTMLElement;
    expect(firstRow).not.toBeNull();

    const spyEvent = jest.fn();
    page.win.addEventListener('click-event', spyEvent);

    firstRow.click();
    expect(spyEvent).toHaveBeenCalled();
  });

  it('emits error event on fetch error', async () => {
    const mockService = {
      fetchTerminalOrders: jest.fn().mockRejectedValue(new Error('Fetch error'))
    };

    const getTerminalOrders = makeGetTerminalOrders({
      id: 'some-id',
      authToken: 'some-auth',
      service: mockService,
      apiOrigin: 'http://localhost:3000'
    });

    const errorEvent = jest.fn();

    const page = await newSpecPage({
      components: components,
      template: () => <terminal-orders-list-core getTerminalOrders={getTerminalOrders} columns={defaultColumnsKeys} onError-event={errorEvent} />,
    });

    await page.waitForChanges();

    expect(errorEvent).toHaveBeenCalledWith(
      expect.objectContaining({
        detail: {
          errorCode: 'fetch-error',
          message: 'Fetch error',
          severity: 'error',
        }
      })
    );
  });
});

describe('terminal-orders-list-core pagination', () => {

  it('updates params and refetches data on pagination interaction', async () => {
    const mockTerminalsService = {
      fetchTerminalOrders: jest.fn().mockResolvedValue(mockOrdersResponse),
    };

    const getTerminalOrders = makeGetTerminalOrders({
      id: '123',
      authToken: '123',
      service: mockTerminalsService,
      apiOrigin: 'http://localhost:3000'
    });

    const page = await newSpecPage({
      components: components,
      template: () => <terminal-orders-list-core getTerminalOrders={getTerminalOrders} columns={defaultColumnsKeys} />,
    });

    await page.waitForChanges();

    // Assuming handleClickNext is accessible and modifies `params` which triggers a re-fetch
    page.rootInstance.handleClickNext('nextCursor');
    await page.waitForChanges();

    // The mock function should be called 2 times: once for the initial load and later after the pagination interaction
    expect(mockTerminalsService.fetchTerminalOrders).toHaveBeenCalledTimes(2);
    const updatedParams = page.rootInstance.pagingParams;
    expect(updatedParams.after_cursor).toBe('nextCursor');
  });
});
