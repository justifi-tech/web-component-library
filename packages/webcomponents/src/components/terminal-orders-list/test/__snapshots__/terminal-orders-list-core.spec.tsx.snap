// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`terminal-orders-list-core displays an error state on failed data fetch 1`] = `
<terminal-orders-list-core>
  <div>
    <div class="table-wrapper">
      <table class="table table-hover" part="table text color font-family background-color">
        <thead class="sticky-top table-head">
          <tr class="table-light text-nowrap">
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The date and time the order was created">
              Created At
            </th>
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The date and time the order was last updated">
              Updated At
            </th>
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The current status of the order">
              Order Status
            </th>
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The number of terminals associated with the order">
              Order Quantity
            </th>
          </tr>
        </thead>
        <tbody class="table-body">
          <tr>
            <td class="error-state" colspan="4" data-test-id="table-error-state" part="table-error text color font-family background-color" style="text-align: center;">
              An unexpected error occurred: Fetch error
            </td>
          </tr>
        </tbody>
        <tfoot class="sticky-bottom">
          <tr class="align-middle table-light">
            <td colspan="4" part="table-cell text color font-family background-color">
              <pagination-menu>
                <nav aria-label="Table pagination" class="d-flex gap-3 justify-content-end">
                  <ul class="m-0 pagination">
                    <li class="disabled page-item">
                      <a class="page-link" href="#" part="button-disabled button text color font-family">
                        <span class="me-1">
                          «
                        </span>
                        <span>
                          Previous
                        </span>
                      </a>
                    </li>
                    <li class="disabled page-item">
                      <a class="page-link" href="#" part="button-disabled button text color font-family">
                        <span>
                          Next
                        </span>
                        <span class="ms-1">
                          »
                        </span>
                      </a>
                    </li>
                  </ul>
                </nav>
              </pagination-menu>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</terminal-orders-list-core>
`;

exports[`terminal-orders-list-core renders properly with fetched data 1`] = `
<terminal-orders-list-core>
  <div>
    <div class="table-wrapper">
      <table class="table table-hover" part="table text color font-family background-color">
        <thead class="sticky-top table-head">
          <tr class="table-light text-nowrap">
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The date and time the order was created">
              Created At
            </th>
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The date and time the order was last updated">
              Updated At
            </th>
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The current status of the order">
              Order Status
            </th>
            <th part="table-head-cell table-cell text color font-family background-color text color font-family background-color" scope="col" title="The number of terminals associated with the order">
              Order Quantity
            </th>
          </tr>
        </thead>
        <tbody class="table-body">
          <tr data-row-entity-id="tord_ABCD1234" data-test-id="table-row">
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Jan 2, 2025
              </div>
              <div class="fw-bold">
                10:30am
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Jan 2, 2025
              </div>
              <div class="fw-bold">
                10:45am
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-primary" part="badge font-family badge-primary" title="This order is submitted">
                Submitted
              </span>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              1
            </td>
          </tr>
          <tr data-row-entity-id="tord_WXYZ5678" data-test-id="table-row">
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Feb 10, 2025
              </div>
              <div class="fw-bold">
                9:20am
              </div>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Feb 10, 2025
              </div>
              <div class="fw-bold">
                9:25am
              </div>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-success" part="badge font-family badge-success" title="This order has been completed">
                Completed
              </span>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              4
            </td>
          </tr>
          <tr data-row-entity-id="tord_DEF55468789" data-test-id="table-row">
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Mar 11, 2025
              </div>
              <div class="fw-bold">
                5:00pm
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Mar 11, 2025
              </div>
              <div class="fw-bold">
                5:05pm
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-secondary" part="badge font-family badge-secondary" title="This order is created">
                Created
              </span>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              2
            </td>
          </tr>
          <tr data-row-entity-id="tord_GHI22168225" data-test-id="table-row">
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Apr 5, 2025
              </div>
              <div class="fw-bold">
                11:50am
              </div>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Apr 5, 2025
              </div>
              <div class="fw-bold">
                12:10pm
              </div>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-success" part="badge font-family badge-success" title="This order has been completed">
                Completed
              </span>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              4
            </td>
          </tr>
          <tr data-row-entity-id="tord_JKL12345678" data-test-id="table-row">
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                May 15, 2025
              </div>
              <div class="fw-bold">
                2:00pm
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                May 15, 2025
              </div>
              <div class="fw-bold">
                2:20pm
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-danger" part="badge font-family badge-danger" title="This order is canceled">
                Canceled
              </span>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              2
            </td>
          </tr>
          <tr data-row-entity-id="tord_PQR12345678" data-test-id="table-row">
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Jun 1, 2025
              </div>
              <div class="fw-bold">
                8:00am
              </div>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Jun 1, 2025
              </div>
              <div class="fw-bold">
                8:15am
              </div>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-info" part="badge font-family badge-info" title="This order is in progress">
                In Progress
              </span>
            </td>
            <td part="table-cell-even table-cell text color font-family background-color text color font-family background-color">
              1
            </td>
          </tr>
          <tr data-row-entity-id="tord_MNO98765432" data-test-id="table-row">
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Jun 20, 2025
              </div>
              <div class="fw-bold">
                8:30am
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <div class="fw-bold">
                Jun 20, 2025
              </div>
              <div class="fw-bold">
                8:45am
              </div>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              <span class="badge bg-warning" part="badge font-family badge-warning" title="This order is on hold">
                On Hold
              </span>
            </td>
            <td part="table-cell-odd table-cell text color font-family background-color text color font-family background-color">
              2
            </td>
          </tr>
        </tbody>
        <tfoot class="sticky-bottom">
          <tr class="align-middle table-light">
            <td colspan="4" part="table-cell text color font-family background-color">
              <pagination-menu>
                <nav aria-label="Table pagination" class="d-flex gap-3 justify-content-end">
                  <ul class="m-0 pagination">
                    <li class="disabled page-item">
                      <a class="page-link" href="#" part="button-disabled button text color font-family">
                        <span class="me-1">
                          «
                        </span>
                        <span>
                          Previous
                        </span>
                      </a>
                    </li>
                    <li class="disabled page-item">
                      <a class="page-link" href="#" part="button-disabled button text color font-family">
                        <span>
                          Next
                        </span>
                        <span class="ms-1">
                          »
                        </span>
                      </a>
                    </li>
                  </ul>
                </nav>
              </pagination-menu>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</terminal-orders-list-core>
`;
